istioctl/cmd/webhook_test.go

```diff
@@ -0,0 +1,974 @@
+			},
+			shouldFail: false,
+		},
+		"valid option 2": {
+			opt: disableCliOptions{
+				disableInjectionWebhook:   true,
+				mutatingWebhookConfigName: "bar",
+			},
+			shouldFail: false,
+		},
+		"valid option 3": {
+			opt: disableCliOptions{
+				disableValidationWebhook:    true,
+				validatingWebhookConfigName: "foo",
+				disableInjectionWebhook:     true,
+				mutatingWebhookConfigName:   "bar",
+			},
+			shouldFail: false,
+		},
+		"invalid option 1": {
+			opt: disableCliOptions{
+				disableValidationWebhook:    true,
+				validatingWebhookConfigName: "",
+			},
+			shouldFail: true,
+		},
+		"invalid option 2": {
+			opt: disableCliOptions{
+				disableInjectionWebhook:   true,
+				mutatingWebhookConfigName: "",
+			},
+			shouldFail: true,
+		},
+		"invalid option 3": {
+			opt: disableCliOptions{
+				disableValidationWebhook: false,
+				disableInjectionWebhook:  false,
+			},
+			shouldFail: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := tc.opt.Validate()
+		if tc.shouldFail {
+			if err == nil {
+				t.Errorf("opt (%v) should have failed", tc.opt)
+			}
+		} else if err != nil {
+			t.Errorf("opt (%v) should not fail, but err: %v", tc.opt, err)
+		}
+	}
+}
+
+func TestStatusCliOptionsValidation(t *testing.T) {
+	testCases := map[string]struct {
+		opt        statusCliOptions
+		shouldFail bool
+	}{
+		"valid option 1": {
+			opt: statusCliOptions{
+				validationWebhook:           true,
+				validatingWebhookConfigName: "foo",
+			},
+			shouldFail: false,
+		},
+		"valid option 2": {
+			opt: statusCliOptions{
+				injectionWebhook:          true,
+				mutatingWebhookConfigName: "bar",
+			},
+			shouldFail: false,
+		},
+		"valid option 3": {
+			opt: statusCliOptions{
+				validationWebhook:           true,
+				validatingWebhookConfigName: "foo",
+				injectionWebhook:            true,
+				mutatingWebhookConfigName:   "bar",
+			},
+			shouldFail: false,
+		},
+		"invalid option 1": {
+			opt: statusCliOptions{
+				validationWebhook:           true,
+				validatingWebhookConfigName: "",
+			},
+			shouldFail: true,
+		},
+		"invalid option 2": {
+			opt: statusCliOptions{
+				injectionWebhook:          true,
+				mutatingWebhookConfigName: "",
+			},
+			shouldFail: true,
+		},
+		"invalid option 3": {
+			opt: statusCliOptions{
+				validationWebhook: false,
+				injectionWebhook:  false,
+			},
+			shouldFail: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := tc.opt.Validate()
+		if tc.shouldFail {
+			if err == nil {
+				t.Errorf("opt (%v) should have failed", tc.opt)
+			}
+		} else if err != nil {
+			t.Errorf("opt (%v) should not fail, but err: %v", tc.opt, err)
+		}
+	}
+}
+
+func TestDisableWebhookConfig(t *testing.T) {
+	testCases := map[string]struct {
+		opt                           disableCliOptions
+		createValidatingWebhookConfig bool
+		createMutatingWebhookConfig   bool
+		shouldFail                    bool
+	}{
+		"disable validating webhook config": {
```

Comment:
The test case keys (names) doesn't seem to be used. Consider simplifying and removing them (`testCases := []string{}`) or printing the name when the test case fails. Using a subtest per test case makes this easier to manage.

Reply:
Done: printing the test case name when the test case fails.
The test case keys also make the test cases more readable.